### PR TITLE
patching failing segfault in test_jit

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes segfault from running `./build/bin/test_jit`

Not totally sure what the root cause is though.

### CODE REVIEW

**Review by kevinstephano — APPROVED** [review]
LGTM

File: third_party/nvfuser/CMakeLists.txt [hunk 1]
```diff
@@ -132,7 +132,8 @@ list(APPEND NVFUSER_SRCS
 )
 
 add_library(${NVFUSER_CODEGEN} SHARED ${NVFUSER_SRCS})
-torch_compile_options(${NVFUSER_CODEGEN})
+# Note: for some reason, torch_compile_options gives us segfaults when we run test_jit
+#torch_compile_options(${NVFUSER_CODEGEN})
```
Comment (naoyam) [inline third_party/nvfuser/CMakeLists.txt:136]: I wonder what the impact of this change would be. Was this the only place compile options were configured? Does removing it mean options like `-Wall` is not used?

**Comment by naoyam** [comment]
I'd say it's more likely we have some real bugs than gcc or nvcc does. Do you have more info on the segfault?

**Comment by jjsjann123** [comment]
I agree that there's likely something very wrong with library load maybe?! I'm a little lost in what's possibly wrong though. The segfault wasn't really pointing me to anything useful. I'll open an issue to track it.

In terms of worries about compiler options, we can certainly add back the missing ones.

**Comment by naoyam** [comment]
Can you please at least keep `-Wall`? I don't know what's happening with the segfault, but we shouldn't leave the options off for a long time as fixing them later would be costly.

**Comment by jjsjann123** [comment]
> Can you please at least keep `-Wall`? I don't know what's happening with the segfault, but we shouldn't leave the options off for a long time as fixing them later would be costly.

Totally agree and I was already on that. Sorry I merged this one as-is

**Pull request merged**